### PR TITLE
Fix: in slot_attention.py the slot initialization parameters were not learnable.

### PR DESCRIPTION
**Pull request by Horse7354**
The slot initialization parameters are now learnable, as the paper says they should be. Also initial variance is scaled appropriately

### CODE REVIEW

**Comment by lucidrains** [comment]
@Horse7354 ohh interesting, could you point me to where / when this change happened in the paper?

**Comment by Horse7354** [comment]
It was always like that in the paper (algorithm1 "We initialize the slots by sampling their initial values
as independent samples from a Gaussian distribution with shared, learnable parameters"), and since we stored slot_mu and slot_sigma as params I assume it was intended to make them learnable. Problem is torch.normal doesn't backprop gradients to it's arguments

**Comment by lucidrains** [comment]
@Horse7354 Sorry, it's been a long time since I worked on this repo. So I think `torch.normal` is differentiable, based on https://github.com/pytorch/pytorch/issues/4620

**Comment by lucidrains** [comment]
@Horse7354 so it was fine the way it is, but let's do it your way to make it more explicit

**Comment by lucidrains** [comment]
@Horse7354 oh actually I'm no longer sure, yea, you may be right. Thank you for catching this!
